src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -602,6 +602,32 @@ def trusted_launch_warning_log(namespace, generation_version, features):
             logger.warning(UPGRADE_SECURITY_HINT)
 
 
+def validate_vm_disk_trusted_launch(namespace, disk_security_profile):
+    if disk_security_profile is None:
+        return
+
+    from ._constants import UPGRADE_SECURITY_HINT
+    security_type = disk_security_profile.security_type if hasattr(disk_security_profile, 'security_type') else None
+    if security_type == 'TrustedLaunch':
+        if namespace.enable_secure_boot is None:
+            namespace.enable_secure_boot = True
+        if namespace.enable_vtpm is None:
+            namespace.enable_vtpm = True
+        namespace.security_type = 'TrustedLaunch'
+    else:
+        logger.warning(UPGRADE_SECURITY_HINT)
+
+
+def validate_image_trusted_launch(namespace):
+    from ._constants import UPGRADE_SECURITY_HINT
+
+    # set securityType to Standard by default if no inputs by end user
+    if namespace.security_type is None:
+        namespace.security_type = 'Standard'
+    if namespace.security_type.lower() != 'trustedlaunch':
```

Comment:
Just a small question, why sometime the judgement logic is ` if security_type == 'TrustedLaunch'`?

Reply:
This param is defined with `arg_type=get_enum_type` in _params.py, and it is converted to `TrustedLaunch` if the input is `trustedlaunch`. Hence `security_type == 'TrustedLaunch'` and `namespace.security_type.lower() != 'trustedlaunch'` both work.
The difference is caused by the different developing style between Taoxu and me hahaha
Will keep consistence.

Reply:
OK, sounds good